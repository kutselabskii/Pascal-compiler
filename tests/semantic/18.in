program abc;

const b = 2;
      c = 10;
var a: real;

begin
a := b * c mod (c * c);
end.