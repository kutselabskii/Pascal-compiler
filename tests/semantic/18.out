|__ program
    |-- abc //Program_heading
    |__ //Block
        |-- //Declaration part
        |   |-- const
        |   |   |-- b
        |   |   |   |__ 2
        |   |   |__ c
        |   |       |__ 10
        |   |-- var
        |   |   |-- a
        |   |   |   |__  : real
        |__ //Statement part
            |__ //Statement sequence
                |-- //Assignment_statement
                |   |-- a
                |   |   |__  : real
                |   |__ //Typecast
                |       |--  : real
                |       |__ 20
